pkg/ingester/ingester_ring.go

```diff
@@ -44,13 +44,16 @@ type RingConfig struct {
 
 	// Config for the ingester lifecycle control
 	ObservePeriod            time.Duration `yaml:"observe_period" category:"advanced"`
-	JoinAfter                time.Duration `yaml:"join_after" category:"advanced"`
+	JoinAfter                time.Duration `yaml:"join_after" category:"advanced" doc:"hidden"` // TODO Deprecated: remove in Mimir 2.4.0.
 	MinReadyDuration         time.Duration `yaml:"min_ready_duration" category:"advanced"`
 	FinalSleep               time.Duration `yaml:"final_sleep" category:"advanced"`
 	ReadinessCheckRingHealth bool          `yaml:"readiness_check_ring_health" category:"advanced"`
 
 	// Injected internally
 	ListenPort int `yaml:"-"`
+
+	// Used only for testing.
+	JoinAfterMock time.Duration `yaml:"-"`
```

Comment:
It's not a mock, and after we rename original field, we can keep using this name.

```suggestion
	JoinAfter time.Duration `yaml:"-"`
```